cli/cli-utils/src/getConfig.ts

```diff
@@ -23,6 +26,20 @@ export async function getConfig (
     ignoreNonAuthSettingsFromLocal: opts.ignoreNonAuthSettingsFromLocal,
   })
   config.cliOptions = cliOptions
+  if (config.configDependencies) {
+    const store = await createOrConnectStoreController(config)
+    await installConfigDeps(config.configDependencies, {
+      registries: config.registries,
+      rootDir: config.lockfileDir ?? config.rootProjectManifestDir,
+      store: store.ctrl,
+    })
+  }
+  if (!config.ignorePnpmfile) {
+    config.hooks = requireHooks(config.lockfileDir ?? config.dir, config)
+    if (config.hooks?.updateConfig) {
+      config = config.hooks.updateConfig(config)
+    }
+  }
 
   if (opts.excludeReporter) {
     delete config.reporter // This is a silly workaround because @pnpm/core expects a function as opts.reporter
```

Comment:
I'm not sure if it's intentional that `config.hooks.updateConfig` can be a promise returning function, but a heads up that this piece of code won't work as expected if `config` is a promise.